compiler/rustc_passes/src/dead.rs

```diff
@@ -509,48 +474,55 @@ impl<'tcx> MarkSymbolVisitor<'tcx> {
         }
     }
 
-    fn solve_rest_impl_items(&mut self, mut unsolved_impl_items: Vec<(hir::ItemId, LocalDefId)>) {
+    fn solve_rest_items(&mut self, mut unsolved_items: Vec<(hir::ItemId, LocalDefId)>) {
```

Comment:
Could this whole method be merged with `mark_live_symbols`? IIUC, we have a worklist loop calling another worklist loop to mark items live. Could we do this with a single level?